src/bind.py

```diff
@@ -330,3 +350,23 @@ def _bind_config_ip_list(self, ips: list[pydantic.IPvAnyAddress]) -> str:
         if not ips:
             return ""
         return f"{';'.join([str(ip) for ip in ips])};"
+
+    def configure(self, config: dict[str, str]) -> None:
+        """Configure the charmed-bind service.
+
+        Args:
+            config: dict of configuration values
+
+        Raises:
+            ConfigureError: when encountering a SnapError
+        """
+        try:
+            cache = snap.SnapCache()
+            charmed_bind = cache[constants.DNS_SNAP_NAME]
+            charmed_bind.set(config)
```

Comment:
Does the set also reload/restart the application if needed?

Reply:
`charmed-bind` is doing that by itself: https://github.com/canonical/charmed-bind/blob/main/snap/hooks/configure#L18

Reply:
Also, is it needed to check the present before the set?